tests/test_formats.py

```diff
@@ -116,28 +116,47 @@ def test_sequencescape22_formatter_does_not_raise_error_on_child_creation():
 
 def test_formatter_breaks_if_not_using_ascii_prefix():
     with pytest.raises(UnsupportedEncodingForPrefix):
-        Sequencescape22Formatter(prefix="MADRILEÑO")
+        Sequencescape22Formatter({"prefix": "MADRILEÑO"})
 
     with pytest.raises(UnsupportedEncodingForPrefix):
-        GenericBarcodeFormatter(prefix="MADRILEÑO")
+        GenericBarcodeFormatter({"prefix": "MADRILEÑO"})
 
 
 def test_sequencescape22_formatter_supports_barcodes_with_suffix():
-    instance = Sequencescape22Formatter(prefix="TEST")
+    instance = Sequencescape22Formatter({"prefix": "TEST"})
     assert instance.barcode(1) == "TEST-1-I"
     assert instance.barcode(2) == "TEST-2-J"
 
-    instance = Sequencescape22Formatter(prefix="TT")
+    instance = Sequencescape22Formatter({"prefix": "TT"})
     assert instance.barcode(1) == "TT-1-O"
     assert instance.barcode(2) == "TT-2-P"
 
-    instance = Sequencescape22Formatter(prefix="TEST")
+    instance = Sequencescape22Formatter({"prefix": "TEST"})
     assert instance.barcode(123) == "TEST-123-V"
     assert instance.barcode(124) == "TEST-124-W"
 
 
+def test_sequencescape22_formatter_supports_barcodes_with_text_code():
+    instance = Sequencescape22Formatter({"prefix": "TEST", "text": "EDU"})
+    assert instance.barcode(1) == "TEST-EDU-1-P"
+    assert instance.barcode(2) == "TEST-EDU-2-Q"
+
+    instance = Sequencescape22Formatter({"prefix": "TT", "text": "T01"})
+    assert instance.barcode(1) == "TT-T01-1-N"
+    assert instance.barcode(2) == "TT-T01-2-O"
+
+    instance = Sequencescape22Formatter({"prefix": "TEST", "text": "T24"})
+    assert instance.barcode(123) == "TEST-T24-123-D"
+    assert instance.barcode(124) == "TEST-T24-124-E"
+
+
+def test_sequencescape22_formatter_supports_barcodes_with_wrong_text_code():
+    with pytest.raises(UnsupportedTextCodeValue):
+        Sequencescape22Formatter({"prefix": "TEST", "text": "T240"})
```

Comment:
does the {1,3} part of the regex apply to the given text here?

Reply:
It is a test for error: the text has more than 3 characters, so it will fail